libraries/core_libs/network/src/tarcap/packets_handlers/vote_packet_handler.cpp

```diff
@@ -13,44 +13,49 @@ VotePacketHandler::VotePacketHandler(std::shared_ptr<PeersState> peers_state,
       vote_mgr_(std::move(vote_mgr)),
       seen_votes_(1000000, 1000) {}
 
-void VotePacketHandler::validatePacketRlpFormat(const PacketData &packet_data) const {
-  if (constexpr size_t required_size = 1; packet_data.rlp_.itemCount() != required_size) {
-    throw InvalidRlpItemsCountException(packet_data.type_str_, packet_data.rlp_.itemCount(), required_size);
+void VotePacketHandler::validatePacketRlpFormat([[maybe_unused]] const PacketData &packet_data) const {
+  auto iteam_count = packet_data.rlp_.itemCount();
+  if (iteam_count == 0 || iteam_count > kMaxVotesInPacket) {
+    throw InvalidRlpItemsCountException(packet_data.type_str_, iteam_count, 1, kMaxVotesInPacket);
```

Comment:
why do we need this new type of exception as you hard corded `1` into it?

Reply:
The exception for checking in range [1, kMaxVotesInPacket]

Reply:
but why can we re-use old one? this add zero addition value in my opinion, especially when `1` is fixed, only relevant information is `iteam_count` but you could use old exception that's there

Reply:
Old exception checks actual size != expected size
The new exception checks actual size is not in range [start, end]
They are different. 

Could you please give me an example how to re-use old one?

Reply:
old exception does not check anything it is thrown if that checks fails, but if you just make more general log that exception is printing you could reuse it as used same way, you just checking for actual `size > expected` but you do not need new exception object for that an especially if you use some hardcoded value e.g. `1` -> zero additional information from that

Reply:
fixed